### PR TITLE
Remove allocation overhead when updating TrackBass amplification data

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Audio/Track/TrackBass.cs [hunk 1]
```diff
@@ -143,6 +143,16 @@ void IBassAudio.UpdateDevice(int deviceIndex)
             Trace.Assert(Bass.LastError == Errors.OK);
         }
 
+        private const int freq_data_length = 256;
+
+        private int currentFreqBufferIndex;
+
+        private readonly float[][] freqDataBuffers =
```
Comment (smoogipoo) [inline osu.Framework/Audio/Track/TrackBass.cs]: What's the reasoning for having two arrays?

Comment (peppy) [reply osu.Framework/Audio/Track/TrackBass.cs]: To maintain the previous behaviour – we shouldn't be updating the data when it's potentially being read by a consumer.

Comment (smoogipoo) [reply osu.Framework/Audio/Track/TrackBass.cs]: This doesn't protect against that. You'd have to somehow copy it to do that, perhaps something like the following would work:

```yaml
class LogoVisualiser:
    var buffer = new float[256];
    
    void Update():
        track.GetFrequencyAmplitudes(buffer);

class Track:
    void UpdateState():
        lock (freqAmplitudeBuffer)
            /* fill buffer */

    void GetFrequencyAmplitures(float[] buffer):
        lock (freqAmplitudeBuffer)
            freqAmplitudeBuffer.CopyTo(buffer);
```

Comment (peppy) [reply osu.Framework/Audio/Track/TrackBass.cs]: right, because we may be more than one update ahead..

**Comment by peppy** [comment]
Any reason this is its own test scene with a weird name ("Scenarios")?

**Comment by peppy** [comment]
This requires a larger API change and isn't a no-effort refactor (`TrackAmplification` should probably not exist), so just gonna leave this one for now.

**Pull request closed without merging**